src/keypress.c

```diff
@@ -215,9 +212,10 @@ void process_keypress(int c) {
             num_lines++;
 
             unsigned int lcx = cx;
-            cursor.x = 0;
             last_cursor_x = 0;
-            process_keypress(KEY_DOWN);
```

Comment:
the call to process_keypress(KEY_DOWN); was inlined because now KEY_DOWN calls syntaxHighlight and the reallocation of lines here caused a segfault in syntaxHighlight . Same thing is for process_keypress(KEY_UP); on line 184